src/main/java/io/supertokens/webserver/api/core/ApiVersionAPI.java

```diff
@@ -45,11 +48,25 @@ protected boolean versionNeeded(HttpServletRequest req) {
     }
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         JsonObject result = new JsonObject();
         JsonArray versions = new JsonArray();
+
+        SemVer maxCDIVersion = getLatestCDIVersion();
```

Comment:
move the logic for the stuff below into the getLatestCDIVersion function

Reply:
Done